new-components/targets/git-clone/Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Used only to build Go binaries.
+FROM golang:1.23.3 AS builder
+
+WORKDIR /wrk
+
+# Copy only go related files.
+COPY go.mod go.sum ./
+COPY vendor ./vendor
+COPY cmd ./cmd
+COPY pkg ./pkg
+
+# Security hardening and building flags for minimal binaries.
+#
+# These CGO_CPPFLAGS help preventing overflows.
+# Add a small overhead at compile time.
+ENV CGO_CPPFLAGS="-D_FORTIFY_SOURCE=2 -fstack-protector-all"
+# Makes memory exploitation harder.
+# Add a small overhead at compile time.
+ENV GOFLAGS="-buildmode=pie"
+ENV CGO_ENABLED=0
+ENV GOOS="linux"
+ENV GOARCH="amd64"
+RUN go build -ldflags "-s -w" -trimpath -o app ./cmd/main.go
+
+# Create a workspace to clone repos to.
+RUN mkdir -p /workspace
+
+# Used to actually run the binary in minimal image.
+FROM gcr.io/distroless/base-debian12:nonroot
```

Comment:
I keep getting errors for missing .so(s) using scratch - if you have suggestions you are welcome!